### DISCUSSION TITLE
discuss: how to handle integration/feature branches

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#981] [Ideas]
#### Description 

When a few stories are dependent and need to be done sequentially.

- Engineer who picks up the first ticket creates integration branch 
- Once the engineer wraps up the task -> they would raise a PR into the integration branch
- After getting an engineer review -> this task's PR would be merged into integration branch 
- The task 1 ticket -> would be moved to PR review state
- Engineer will drop an update on the tickets stating this task is merged into integration branch 
- product / design would review the integration branch for task 1 and drop feedback on the integration PR 

NOTE: 
- No PRs should be merged without tests into the integration branch 
- If changes are requested on a task -> the team has to branch off the integration branch and make those changes and merged it into integration branch after engineering review
- The integration PR will finally be merged into the main branch after product + design +engineering approve PR 

Here is an example :
https://github.com/gitcoinco/grants-round/pull/982

**Comment by vacekj** [comment]
Closing as stale.
